libdnf/sack/Solution.cpp

```diff
@@ -25,58 +25,56 @@ namespace libdnf {
 
 bool
 Solution::getBestSolution(const char * subject, DnfSack* sack, HyForm * forms, bool icase,
-    bool with_nevra, bool with_provides, bool with_filenames, bool with_src)
+    bool with_nevra, bool with_provides, bool with_filenames, bool with_src,
+    Query::ExcludeFlags exclude_flags)
 {
+    query.reset(new Query(sack, exclude_flags));
     nevra.reset();
-    Query baseQuery(sack);
+    Query baseQuery(*query);
+
     if (!with_src) {
         baseQuery.addFilter(HY_PKG_ARCH, HY_NEQ, "src");
     }
-    baseQuery.apply();
-    std::unique_ptr<Query> queryCandidate(new Query(baseQuery));
+
+    query->addFilter(HY_PKG_EMPTY, HY_EQ, 1);
```

Comment:
This is unneeded step. query == baseQuery therefore
query->addFilter(HY_PKG_EMPTY, HY_EQ, 1)
query->queryUnion(baseQuery);

result in same result like 
query->queryUnion(baseQuery);


Then line query->addFilter(HY_PKG_EMPTY, HY_EQ, 1) please move to line 78.

Reply:
The propose construct has a better performance because it allows to skip addFilter(HY_PKG_EMPTY, HY_EQ, 1) step when matches found.

Reply:
Fixed.